### DISCUSSION TITLE
Default permissions (ACL) of "Microsoft.WindowsTerminal" folder in "WindowsApps".

### DISCUSSION THREAD

**Discussion by Fooughhy** [microsoft/terminal#13511] [Q&A]
### Windows build number

10.0.19044.0

### Steps to reproduce

Mess about with "WindowsApps" folder permissions without backing them up before...

### Actual Behavior

I would like to know if anyone could provide all the default permissions that the "%ProgramFiles%\WindowsApps" folder has, as well as the "Microsoft.WindowsTerminal" folders (I think there are two, right?). I have had an unbelievable amount of issues with Windows Store apps lately (started with only Xbox Games Pass games which stopped working), and as a final resort I tried to mess with the permissions of the folders to be able to check that the "AppxManifest.xml" was there (as there were errors indicating it wasn't, at some point at least). These issues have also prevented me from using RestorePoints which I had made once I actually got into a working state.

But now the Xbox Game Pass games are working, but for some reason there have been multiple UWP apps that doesn't work if they are started from the store (or from the context menu either, actually). And one of them is the Windows Terminal.

I received an error indicating that the "Parameter is incorrect" for "wt.exe". This led me to [this](https://superuser.com/questions/1288014/reset-default-acls-for-c-program-files-windowsapps) thread, amd the answer by [Gidsik](https://superuser.com/users/1659735/gidsik) telling me to run the command "cacls "%programfiles%\WindowsApps" /s:"D:PAI(A;;FA;;;S-1-5-80-956008885-3418522649-1831038044-1853292631-2271478464)(A;OICIIO;GA;;;S-1-5-80-956008885-3418522649-1831038044-1853292631-2271478464)(A;;0x1200a9;;;S-1-15-3-1024-3635283841-2530182609-996808640-1887759898-3848208603-3313616867-983405619-2501854204)(A;OICIIO;GXGR;;;S-1-15-3-1024-3635283841-2530182609-996808640-1887759898-3848208603-3313616867-983405619-2501854204)(A;;FA;;;SY)(A;OICIIO;GA;;;SY)(A;CI;0x1200a9;;;BA)(A;OICI;0x1200a9;;;LS)(A;OICI;0x1200a9;;;NS)(A;OICI;0x1200a9;;;RC)(XA;;0x1200a9;;;BU;(Exists WIN://SYSAPPID))"" to set the correct permissions for the "WindowsApps" folder. But as that folder seems to work (other UWP apps work) I decided only to run it on the Terminal folders inside. This now means that I can't open the terminal at all anymore...

So if anyone could provide the ACLs that I can use to run that command again for the WindowsTerminal folder, that'd be hugely appreciated.

**Comment by zadjii-msft** [comment]
> Mess about with "WindowsApps" folder permissions without backing them up before

_y tho_

For real, that will fuck your machine up. Don't do that, and if there's something that suggested you do that, I'd love to reach out to the origin and find out WHY.

**Comment by zadjii-msft** [comment]
@DHowett MIGHT have a way to repair this.

**Comment by DHowett** [comment]
Unfortunately, there's no general way to un-mess the permissions in that folder. It doesn't exactly have "default" permissions: everything inside is ACLed based on a _bunch_ of different things[1].

There's one script _which we do not recommend running on a production system, unless you have no other option_ called "Windows Apps Unf**er". I won't be providing a link to it.

The _best and least destructive_ way to fix the permissions in that folder is honestly to do an in-place upgrade to the same version of Windows you already have. That is: if you're on "Windows 11 build 22000," grab an ISO file for that same **exact** build and perform an upgrade that "keeps personal files/settings and apps."

[1]: Packages can share files (they're deduplicated), so sometimes the ACLs for one package include another package's ID; there's this thing called "hosted runtimes"; some packages are hosted outside this folder but have permissions to let those outside applications access this folder, etc.

**Comment by Fooughhy** [comment]
Hi!

Yes, I know it is potentially very destructive to mess around with that folder but I felt like I had no other choice. I created [this post](https://answers.microsoft.com/en-us/xbox/forum/all/im-receiving-error-0x80004005-when-opening-games/df6111dd-b128-489b-986c-3d408e6ffd5e) to help fix my initial issue, and I even did a In-Place upgrade there. And that worked, but for some unknown reason it reverted after a couple of reboots.

@DHowett Oh, that was much more complicated than I thought. I had gotten the impression that there was no single script to use for everything but I thought I could at least copy the permissions from someone else. But that isn't possible then?
So scripts like the following seems to not fix this, is the impression I have gotten.
> icacls "C:\Program Files\WindowsApps" /reset /t /c /q

But how do I do an In-Place Upgrade for the exact same build? Don't I need to create the ISO myself then? From my faulty system?

**Comment by xarathos86** [comment]
Use this:
https://github.com/AgentRev/WindowsAppsUnfukker
I'm a Sysadmin, tried most stuff, this works as a charm!
I've tried everything on this folder trying to fix it (just for fun) after cloning an SSD to a new one).
Keep this tool in a safe place.

**Reply by phantai987654321987654321-kalilinux** [reply]
Hi Xarathos86,
I just make sure that I won't do more st fuckup because I don't understand the problem before try to do your innstruction
For accessing WindowsApps, I changed the permission following a common construction. But instead of confirming for the icon named Administrator with the down arrow, I permitted the icon named Administrator with 2 persons (See attached picture)
Then I try to reverse my action by below script in CMD
icacls "C:\Program Files\WindowsApps" /reset /t /c /q
It didn't help after restart, so I try with 3 other scripts, respectively
TAKEOWN /F "C:\Program Files\WindowsApps" /R /D N
icacls "C:\Program Files\WindowsApps" /reset /t /c /q
icacls "C:\Program Files\WindowsApps" /setowner "NT SERVICE\TrustedInstaller" /t /c
The error is still parameter is not correct
Know that my laptop uses Window10 ver 10.0.19045
<img width="249" height="58" alt="image" src="https://github.com/user-attachments/assets/efd1b7e9-8acb-44eb-a9aa-54f4893e134e" />
